catalog/camel-route-parser/src/main/java/org/apache/camel/parser/helper/ParserCommon.java

```diff
@@ -118,8 +118,8 @@ static FieldSource<JavaClassSource> getField(JavaClassSource clazz, Block block,
     public static boolean isNumericOperator(JavaClassSource clazz, Block block, Expression expression) {
         if (expression instanceof NumberLiteral) {
             return true;
-        } else if (expression instanceof SimpleName) {
-            FieldSource<JavaClassSource> field = getField(clazz, block, (SimpleName) expression);
+        } else if (expression instanceof SimpleName simpleName) {
+            FieldSource<JavaClassSource> field = getField(clazz, block, simpleName);
```

Comment:
Also, this part here implies that there is something more other than tests. Maybe adjust the PR description, commit message and/or reference a Jira ticket if you are providing a fix.